site/src/pages/UserSettingsPage/OAuth2ProviderPage/OAuth2ProviderPage.tsx

```diff
@@ -0,0 +1,59 @@
+import { type FC, useState } from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { getErrorMessage } from "api/errors";
+import { getApps, revokeApp } from "api/queries/oauth2";
+import type * as TypesGen from "api/typesGenerated";
+import { DeleteDialog } from "components/Dialogs/DeleteDialog/DeleteDialog";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import { useMe } from "hooks";
+import { Section } from "../Section";
+import OAuth2ProviderPageView from "./OAuth2ProviderPageView";
+
+const OAuth2ProviderPage: FC = () => {
+  const me = useMe();
+  const queryClient = useQueryClient();
+  const userOAuth2AppsQuery = useQuery(getApps(me.id));
+  const revokeAppMutation = useMutation(revokeApp(queryClient, me.id));
+  const [appToRevoke, setAppToRevoke] = useState<TypesGen.OAuth2ProviderApp>();
```

Comment:
I'm worried about duplicated state here – should the tracking state be the whole app, or only the ID for the app?

My worry is that if we hold an entire app in two places, if the mutation state removes it from the server cache for whatever reason, we'd still have the stale app data held somewhere else. At least with an ID, it'd be harder to do that:

```ts
const userOAuth2AppsQuery = useQuery(getApps(me.id));
const [idToRevoke, setIdToRevoke] = useState<string>();

const appToRevoke = userOAuth2AppsQuery.data?.find((app) => app.id === idToRevoke);
```
Might even be worth adding an [inline state sync](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) to reset the ID state if that happens:
```ts
const userOAuth2AppsQuery = useQuery(getApps(me.id));
const [idToRevoke, setIdToRevoke] = useState<string>();

const appToRevoke = userOAuth2AppsQuery.data?.find((app) => app.id === idToRevoke);

// This is safe to place in the render, because it doesn't run every single time
if (appToRevoke === undefined && typeof idToRevoke === "string') {
  setAppIdToRevoke(undefined);
}
```

Reply:
Oh good catch, I had not considered that.  I will swap to the ID!